cfclient/cfapps/manager.go

```diff
@@ -118,7 +118,7 @@ func (c *CFAppManager) FetchApp(a *CFApp) error {
 	}
 
 	c.clientLock.RLock()
-	result, err := c.client.GetAppByGuid(a.GUID)
+	result, err := c.client.GetAppByGuidNoInlineCall(a.GUID)
```

Comment:
It would be good to know the impact of this change, is this call available for tas versions >2.4 ? what is the difference vs GetAppGuid?

Reply:
These methods are defined in go-cfclient -> apps.go.

GetAppByGuid calls /v2/apps/<GUID>?inline-relations-depth=2 to get all data we need. The inline-relations-depth call is an expensive call that can cause impact at the cloud controller.

GetAppByGuidNoInlineCall uses three separate, lighter weight API calls to retrieve this information instead.
- a GET on /v2/apps/<GUID>
- a GET to retrieve the application's space details
- a GET to retrieve the application's org details.

Even though there are more total calls, this should reduce impact on the cloud controller API.

Reply:
Integration tests were missing due to the new endpoints, I added a commit fixing those